internal/server/utils.go

```
@@ -0,0 +1,26 @@
+package server
+
+import (
+	"crypto/rsa"
+	"crypto/sha1"
+	"crypto/x509"
+	"encoding/asn1"
+)
+
+// generateSKI generates the Subject Key Identifier (SKI) for the given private key.
+// The SKI is the SHA-1 hash of the public key.
+// The SKI is used to identify the public key in the certificate and is used in the Authority Key Identifier extension.
+// The SKI is necessary for CRL signing.
+func generateSKI(priv *rsa.PrivateKey) ([]byte, error) {
+	pubKeyBytes, _ := x509.MarshalPKIXPublicKey(&priv.PublicKey)
```

Why ignore the error coming out of `MarshalPKIXPublicKey`? We should likely capture it and return it

---

same reason as [this](https://github.com/canonical/notary/pull/204#discussion_r2016341916) comment.

---

I still don't get the added value of not checking for the error. The callee doesn't have information about the caller, why assume anything about it?

---

not checking for the error is the null value, where we discard errors that wouldn't effect the program (in this case because this is unreachable code). I don't see the value in having detailed and informative error messages and handling in code that people can't see, even if that may be possible in the future. I would consider it the responsibility of the developer to not misuse internal functions.

justifications aside, I think there's a better common ground way to do this, I've responded to it above.